### PR TITLE
Semver metadata variables

### PR DESCRIPTION
**Pull request by splatteredbits**
* Added a couple of new Whiskey variables.
* Added the ability to call methods/properties on Whiskey variable objects.

### CODE REVIEW

**Comment by splatteredbits** [comment]
@jlariono @jeffprice6 Will you review this pull request?

File: Whiskey/en-US/about_Whiskey_Variables.help.txt [hunk 1]
```diff
@@ -0,0 +1,90 @@
+TOPIC
+    about_Whiskey_Variables
+
+SHORT DESCRIPTION
+    This topic describes how to use Whiskey variables in your whiskey.yml file
+    and what built-in variables are available.
+
+LONG DESCRIPTION
+
+    ## Using
```
Comment (jlariono) [inline Whiskey/en-US/about_Whiskey_Variables.help.txt]: Nit: This heading looks like a word or sentence was forgotten. `Using Variables` sounds better.

File: Whiskey/Functions/Resolve-WhiskeyVariable.ps1 [hunk 2]
```diff
@@ -93,26 +93,40 @@ function Resolve-WhiskeyVariable
 
         $version = $Context.Version
         $buildInfo = $Context.BuildMetadata;
+        $sem1Version = ''
+        if( $version.SemVer1 )
+        {
+            $sem1Version = '{0}.{1}.{2}' -f $version.SemVer1.Major,$version.SemVer1.Minor,$version.SemVer1.Patch;
```
Comment (jlariono) [inline Whiskey/Functions/Resolve-WhiskeyVariable.ps1]: Nit: Unnecessary semicolon.

File: Whiskey/Functions/Resolve-WhiskeyVariable.ps1 [hunk 3]
```diff
@@ -93,26 +93,40 @@ function Resolve-WhiskeyVariable
...
+        $sem1Version = ''
+        if( $version.SemVer1 )
+        {
+            $sem1Version = '{0}.{1}.{2}' -f $version.SemVer1.Major,$version.SemVer1.Minor,$version.SemVer1.Patch;
+        }
+
+        $sem2Version = ''
+        if( $version.SemVer2 )
+        {
+            $sem2Version = '{0}.{1}.{2}' -f $version.SemVer2.Major,$version.SemVer2.Minor,$version.SemVer2.Patch;
```
Comment (jlariono) [inline Whiskey/Functions/Resolve-WhiskeyVariable.ps1]: Nit: Unnecessary semicolon.

File: Whiskey/Functions/Resolve-WhiskeyVariable.ps1 [hunk 4]
```diff
@@ -175,8 +189,89 @@ function Resolve-WhiskeyVariable
                 }
             }
 
-            $needleEnd = $haystack.IndexOf(')', $needleStart)
+            $needleEnd = $needleStart + 2
+            $depth = 0
+            while( $needleEnd -lt $haystack.Length )
```
Comment (jlariono) [inline Whiskey/Functions/Resolve-WhiskeyVariable.ps1]: This whole `while` section could use some comments describing what each section is doing. As it stands right now, I'd have a hell of a time debugging some code in this loop.

File: Test/Resolve-WhiskeyVariable.Tests.ps1 [hunk 5]
```diff
@@ -295,3 +295,194 @@ Describe 'Resolve-WhiskeyVariable.when value is empty' {
...
+    WhenResolving '$(WHISKEY_SCM_URI.Host)'
+    ThenValueIs 'example.com'
+}
+
+Describe 'Resolve-WhiskeyVariable.when variable not terminated' {
+    Init
+    $value = "The quick brown fox jumped over the lazy dog."
+    GivenVariable 'Fubar' $value
+    WhenResolving '$(Fubar' -ErrorAction SilentlyContinue
+    ThenValueIs '$(Fubar'
```
Comment (jlariono) [inline Test/Resolve-WhiskeyVariable.Tests.ps1]: Nit: Forgot to add a `ThenErrorIs` test.

File: Test/Resolve-WhiskeyVariable.Tests.ps1 [hunk 6]
```diff
@@ -295,3 +295,194 @@ Describe 'Resolve-WhiskeyVariable.when value is empty' {
...
+    $context.BuildMetadata.ScmCommitID = 'deadbeedeadbeedeadbeedeadbeedeadbeedeadb'
+    WhenResolving '$(WHISKEY_SCM_COMMIT_ID.Substring(0,7))'
+    ThenValueIs 'deadbee'
+}
+
+Describe 'Resolve-WhiskeyVariable.when method call is invalid' {
+    Init
+    GivenVariable 'Fubar' 'g'
+    WhenResolving '$(Fubar.Substring(0,7))' -ErrorAction SilentlyContinue
+    ThenValueIs '$(Fubar.Substring(0,7))'
```
Comment (jlariono) [inline Test/Resolve-WhiskeyVariable.Tests.ps1]: Nit: Forgot to add a `ThenErrorIs` test.

**Comment by splatteredbits** [comment]
@jlariono Changes made. Ready for a re-review.

**Comment by splatteredbits** [comment]
@jlariono Can I get another review?

**Pull request merged**